test/active_job/queue_adapters/async_job_adapter.rb

```diff
@@ -93,7 +93,39 @@
 			# This test verifies that the job gets dispatched successfully at a specific time
 			# The Sync wrapper is tested implicitly by the successful enqueueing
 			expect(dispatcher.queues["default"]).to be_a(Object)
-			adapter.enqueue_at(job, timestamp)
+			adapter.enqueue_at(job, timestamp.to_f)
+		end
+	end
+	
+	with "job with :wait option" do
+		let(:adapter) {subject.new(dispatcher)}
+		let(:job) {TestJob.new}
+		let(:wait) {0.second}
+		
+		before do
+			dispatcher.queues["default"] = queue
+			job.set(wait:)
+		end
```

Comment:
The other spec tests that calling `adapter.enqueue_at(job, timestamp.to_f)` without explicitly setting `scheduled_at` on the job via `set` will successfully set `scheduled_at`.

This new spec tests that setting it on the job as you're supposed to (via `set`) will work.

We might not want to support the former test, but again, leaving it just in case we want to guarantee that.